library/x509_csr.c

```diff
@@ -104,61 +103,67 @@ static int x509_csr_parse_extensions(mbedtls_x509_csr *csr,
         /* Data should be octet string type */
         if ((ret = mbedtls_asn1_get_tag(p, end_ext_data, &len,
                                         MBEDTLS_ASN1_OCTET_STRING)) != 0) {
-            return MBEDTLS_ERR_X509_INVALID_EXTENSIONS + ret;
+            return MBEDTLS_ERROR_ADD(MBEDTLS_ERR_X509_INVALID_EXTENSIONS, ret);
         }
+
         if (*p + len != end_ext_data) {
-            return MBEDTLS_ERR_X509_INVALID_EXTENSIONS +
-                   MBEDTLS_ERR_ASN1_LENGTH_MISMATCH;
+            return MBEDTLS_ERROR_ADD(MBEDTLS_ERR_X509_INVALID_EXTENSIONS,
+                                     MBEDTLS_ERR_ASN1_LENGTH_MISMATCH);
         }
 
         /*
          * Detect supported extensions
          */
         ret = mbedtls_oid_get_x509_ext_type(&extn_oid, &ext_type);
 
-        if (ret == 0) {
-            /* Forbid repeated extensions */
-            if ((csr->ext_types & ext_type) != 0) {
-                return MBEDTLS_ERR_X509_INVALID_EXTENSIONS;
-            }
+        if (ret != 0) {
```

Comment:
Should we error on unsupported extensions? That wasn't the behaviour before, and I don't think we want to drop any csr that has just one thing that's not critical and we don't support. Does the RFC say anything about it? See `x509_get_crt_ext` for reference.

Reply:
Good point need to check this. 
There is a test for unsupported extension: `X509 CSR ASN.1 (extensions: invalid extension type data)`.

Reply:
There is not much about unsupported extensions (https://datatracker.ietf.org/doc/html/rfc2511) but as they are optional I and not all can be supported by us I agree we should skip them instead of returning error.